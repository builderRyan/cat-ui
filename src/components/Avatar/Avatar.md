Example

```jsx
class Example extends React.Component {
  constructor (props) {
    super(props)
    this.state = {}
  }

  render () {
    return (<div className='avatar-box'>
      <Avatar src={'http://cdn.himalaya.com/8c205ad6954541c99f7fd77008ff6479.jpg?x-oss-process=image/resize,w_300,h_300&auth_key=4102416000-1234-0-b5602166a98343b7e7f8f58d645d6d3e'} size='small'>
        test
      </Avatar>

      <style>
        {`
          .avatar-box {
            padding: 5px;
          }
        `}
      </style>
    </div>
    )
  }
}

<Example />
```
