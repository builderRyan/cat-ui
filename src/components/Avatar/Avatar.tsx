import * as React from 'react';
import { Component } from 'react';
import classNames from 'classnames';

export interface AvatarProps {
  /** 类名 */
  className?: string;
  /** 样式 */
  style?: object;
  /** 尺寸 */
  size?: number | 'large' | 'medium' | 'small';

  /** 形状 */
  shape?: 'circle' | 'square';

  /** 图片地址 */
  src: string;
  /** 图片名称 */
  alt?: string;
  /** src扩展，会根据ua请求相应的资源 */
  srcSet?: string;

  fit?: 'fill' | 'contain' | 'cover' | 'none' | 'scale-down';

  prefix?: string;

  onError?: (e: any) => boolean;
}

/**
 * Avatar
 */
class Avatar extends Component<AvatarProps, any> {
  handleImgLoadError = (e: any) => {
    this.props.onError(e);
  };

  render() {
    const {
      prefix = 'hmly-avatar',
      shape,
      className,
      children,
      src,
      alt,
      srcSet,
      size,
      onError,
      ...restProps
    } = this.props;
    if (!src) return null;

    const sizeCls = classNames({
      [`${prefix}-lg`]: size === 'large',
      [`${prefix}-sm`]: size === 'small'
    });

    const classes = classNames(prefix, className, sizeCls, {
      [`${prefix}-${shape}`]: shape,
      [`${prefix}-image`]: src
    });

    return (
      <span className={classes} prefix={prefix} {...restProps}>
        <img
          src={src}
          alt={alt}
          srcSet={srcSet}
          onError={this.handleImgLoadError}
        />
      </span>
    );
  }
}

export default Avatar;
