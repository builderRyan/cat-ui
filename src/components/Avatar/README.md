# Avatar 头像

## Example

```jsx
<Avatar
  src='https://raw.githubusercontent.com/sialvsic/static-host/master/public/laptop.png'
  size='large'
/>
```

## API

|    属性     |         类型          |    默认值    | 必选  |                描述                 |
| :---------: | :-------------------: | :----------: | :---: | :---------------------------------: |
|  className  |       `string`        |      -       | false |                类名                 |
|    style    |       `object`        |      -       | false |              行内样式               |
|    size     |   `number | string`   |   `medium`   | false | 尺寸，可选 `large` `medium` `small` |
|    shape    |       `string`        |   `square`   | false |    形状，可选 `circle` `square`     |
|     src     |       `string`        |      -       | false |              图片地址               |
| fallbackSrc |       `string`        |      -       | false |            容错图片地址             |
|     alt     |       `string`        |      -       | false |              图片名称               |
|   srcSet    |       `string`        |      -       | false | src 扩展，会根据 ua 请求相应的资源  |
|    sizes    |       `string`        |      -       | false |   响应式图片，和 srcSet 一起使用    |
|   prefix    |       `string`        | `cat-avatar` | false |              样式前缀               |
|   onError   | `(e: any) => boolean` |      -       | false |           请求出错的回调            |

## Todo

- 支持文字展示
