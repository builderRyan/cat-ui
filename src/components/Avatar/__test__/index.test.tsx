import * as React from 'react';
import { shallow } from 'enzyme';
import Avatar from '../index';

describe('Avatar', () => {
  test('renders correctly', () => {
    const wrapper = shallow(
      <Avatar src='https://raw.githubusercontent.com/sialvsic/static-host/master/public/laptop.png' />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('renders size={20}', () => {
    const wrapper = shallow(
      <Avatar
        size={50}
        src='https://raw.githubusercontent.com/sialvsic/static-host/master/public/laptop.png'
      />
    );
    const style = wrapper.find('.cat-avatar').props().style;
    expect(style).toEqual({ width: 50, height: 50, lineHeight: 50 });
  });
});
