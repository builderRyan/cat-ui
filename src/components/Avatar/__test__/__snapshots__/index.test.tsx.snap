// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders src 1`] = `
<span
  className="cat-avatar cat-avatar-square cat-avatar-image"
  prefix="cat-avatar"
  style={Object {}}
>
  <img
    onError={[Function]}
    src="https://raw.githubusercontent.com/sialvsic/static-host/master/public/laptop.png"
  />
</span>
`;
